### DISCUSSION TITLE
React + React Native vs React Native Web

### DISCUSSION THREAD

**Discussion by Fiorim** [frontendbr/forum#1549] [Perguntas e Respostas]
Fala pessoal, tudo bom?

Qual estratégia vocês me indicariam pra desenvolver um app web e mobile usando React?

Minha equipe já optou por usar React e pelo que vi, tem diversas formas de compartilhar código entre a versão web e mobile.

A questão é: esse "compartilhamento" é recomendável ou é melhor ter 2 apps completamente independentes pra não dificultar a manutenção e perder em performance?

Alguma opinião sobre o  Reactive Native Web?

Obrigado!

**Comment by wilsonneto-swe** [comment]
Aqui na empresa a gente está adotando React, mas pelo que vimos vamos compartilhar apenas o conceito, estrutura e dinâmica do react/es. A escrita vai ser personalizada pra cada, até porque tem muito componente e feature no native que no browser não daria certo, além da experiência do usuário que muda bastante também.

**Comment by Fiorim** [comment]
> Aqui na empresa a gente está adotando React, mas pelo que vimos vamos compartilhar apenas o conceito, estrutura e dinâmica do react/es. A escrita vai ser personalizada pra cada, até porque tem muito componente e feature no native que no browser não daria certo, além da experiência do usuário que muda bastante também.

Valeu pela resposta @wilsonneto-dev , pelo que estou vendo não parece ser muito comum misturar web e mobile.

**Comment by gss-patricia** [comment]
Aqui onde trabalho, temos o app com RN e o portal em React, ambos compartilhando helpers, Sagas, entre outros, essa camada é a "shared", a regra de negocio, tanto para um, quanto para outro, é a mesma, portanto, temos que implementar a logica pensando na aplicação entre as duas plataformas. A experiência do usuário é um pouco diferente em cada plataforma.

**Comment by wilsonneto-swe** [comment]
Isso, verdade @gss-patricia, aqui também não reescrevemos tudo, tentamos sempre colocar a lógica totalmente desacoplada, assim vamos mudar mesmo componentes de UI / UX, e alguns reaproveitamos também.

Mas reescrevemos praticamente 85% rs
